### PR TITLE
reformat matrix_rank #1649

### PR DESCRIPTION
**Pull request by hyeonukim**
Updated Functional Examples

### CODE REVIEW

**Comment by rush2406** [comment]
Hi @hyeonukim
Thanks for your work! However, there are a few things to be done:
- Can you please reference the issue linked to this PR? [issue](https://github.com/unifyai/ivy/issues/1649)
- Add an example using ivy.Container in [file](https://github.com/unifyai/ivy/blob/master/ivy/functional/ivy/linear_algebra.py)
- Add an example for the array instance method [here](https://github.com/unifyai/ivy/blob/master/ivy/array/linear_algebra.py#L90) 
- Implement the static method and container method for `matrix_rank`. Also add examples for both these methods. For reference, you can check [https://github.com/unifyai/ivy/blob/master/ivy/container/linear_algebra.py#L99](here)

Let me know if there are any further questions!

**Comment by hyeonukim** [comment]
Hello @rush2406 

Thank you for such a fast review on my PR!

I had few questions regarding your reply:

1. I wanted to refer to the link that you have but it seems like the link you have given is not working right now. Is it possible to get a link for implementing the static method and container method for my functionality again?

2. Is there a way to manually reference the issue linking to this PR? I have tried to edit my issues using the tab on right side but it seems like I have no authorization

Thanks again for the review!

**Comment by rush2406** [comment]
Hi @hyeonukim 
1. Sorry, the updated link is [here](https://github.com/unifyai/ivy/blob/e45803551012432404ef864980ef8c6490dc1f77/ivy/container/linear_algebra.py#L43). You can follow the changes done for `cholesky` function
2. #1649 , okay I have referenced it here

**Review by rush2406 — CHANGES_REQUESTED** [review]
Hi @hyeonukim 
Can you please include the container static and instance examples as explained?

File: ivy/array/linear_algebra.py [hunk 1]
```diff
@@ -94,6 +94,33 @@ def matrix_rank(
         out: Optional[ivy.Array] = None,
     ) -> ivy.Array:
         return ivy.matrix_rank(self._data, rtol, out=out)
+        """
+        ivy.Array instance method variant of ivy.matrix_rank. This method returns 
+        the rank (i.e., number of non-zero singular values) of a matrix (or a stack of
+        matrices).
+
```
Comment (rush2406) [inline ivy/array/linear_algebra.py:101]: Hi @hyeonukim 
Can you add the parameters and return docstrings as well to complete this PR? You can follow [this](https://github.com/unifyai/ivy/blob/f6738a28ca5d997f70ef6a8e4be7d367d35bf32b/ivy/array/linear_algebra.py#L31)

File: ivy/container/linear_algebra.py [hunk 2]
```diff
@@ -134,3 +134,88 @@ def cholesky(
...
+        map_sequences: bool = False,
+        *,
+        out: Optional[ivy.Container] = None,
+    ) -> ivy.Container:
+
+        """
+        ivy.Container static method variant of ivy.matrix_rank.
+        This method returns the rank (i.e., number of non-zero singular values) of a matrix (or a stack of
+        matrices).
+
```
Comment (rush2406) [inline ivy/container/linear_algebra.py:154]: same here, please add the parameters and return docstrings, [example](https://github.com/unifyai/ivy/blob/f6738a28ca5d997f70ef6a8e4be7d367d35bf32b/ivy/container/linear_algebra.py#L59)

File: ivy/container/linear_algebra.py [hunk 3]
```diff
@@ -134,3 +134,88 @@ def cholesky(
...
+        map_sequences: bool = False,
+        *,
+        out: Optional[ivy.Container] = None,
+    ) -> ivy.Container:
+
+        """
+        ivy.Container instance method variant of ivy.matrix_rank.
+        This method returns the rank (i.e., number of non-zero singular values) of a matrix (or a stack of
+        matrices).
+
```
Comment (rush2406) [inline ivy/container/linear_algebra.py:197]: same here, please add the parameters and return docstrings, [example](https://github.com/unifyai/ivy/blob/f6738a28ca5d997f70ef6a8e4be7d367d35bf32b/ivy/container/linear_algebra.py#L147)

**Review by rush2406 — CHANGES_REQUESTED** [review]
Hi @hyeonukim 
can you please complete docstrings as mentioned in the comments and then re-request review when done

**Comment by hyeonukim** [comment]
Hello, @rush2406

I have edited all the things you have requested for could you check it over again?

Thank you!

File: ivy/container/linear_algebra.py [hunk 4]
```diff
@@ -134,3 +134,174 @@ def cholesky(
...
+        >>> y = x.matrix_rank()
+        >>> print(y)
+        {
+            a: ivy.array(2.),
+            b: ivy.array(1.)
+        }
+        
+        """ 
+
+        return self.matrix_rank(
```
Comment (rush2406) [inline ivy/container/linear_algebra.py:299]: This should be `self.static_matrix_rank`, which is currently causing the docstrings tests to fail

**Review by rush2406 — CHANGES_REQUESTED** [review]
Hi @hyeonukim 
Looks great! thanks for the changes. There's a change required in the container file, see the comment below!  Also
1. Can you please pull the master changes because there are conflicts now.
2. The `lint` and `numy-docstring` tests are failing. These are just formatting issues related to spaces, new lines etc. Please fix them as well.
Please request a review once done!

**Comment by rush2406** [comment]
@hyeonukim you need not do formatting for `cholesky`, can you please restore the blank lines you've removed, it's causing the `numpy-docstrings` to fail. Remaining looks good, please request a review once done and it's ready to merge!

**Comment by hyeonukim** [comment]
Hi @rush2406 
I do not know how that happened with cholesky functions as I did not touch them, sorry about that.
I did fix all the issues could you do a final review?
Thanks!

**Comment by rush2406** [comment]
@hyeonukim thanks for the changes! I have merged the PR.

**Pull request merged**